exploration/0002-expression-attributes.md

```diff
@@ -38,18 +38,59 @@ At least the following expression attributes should be considered:
 
   - `fallback` — A value to use instead of the default fallback,
     should the expression's primary formatting fail in some way.
+
+    > Example that would format as "Hello world" if `$place` is not defined:
+    >
+    > ```
+    > {Hello {$place @fallback=world}}
+    > ```
+
   - `locale` — An override for the locale used to format the expression.
     Should be expressed as a non-empty sequence of BCP 47 language codes.
+
+    > Example embedding a French literal in an English message:
+    >
+    > ```
+    > {In French, "{|bonjour| @locale=fr}" is a greeting}
```

Comment:
Is this use-case also solved by spans?

```
{In French, {+span locale=fr}bonjour{-span} is a greeting}
```

`+span`/`-span` doesn't even have to mean HTML `span` here. It's just a start and an end of a span of text that should have a different locale property. The runtime can then decide how to express it such that the locale is isolated properly.

Reply:
@stasm I don't think so. The spans don't affect their interior text? Recall that in formatToParts the `MessageValue` has direction and language properties for this. Although, your formulation has a certain attraction, as "bonjour" is just some text. One can imagine this:

```
In French {+g @locale=fr @translate=no}bonjour{-g} is a greeting
```

(using `g` for `xliff:g`)